internal/blocksync/reactor.go

```diff
@@ -76,7 +76,7 @@ type Reactor struct {
 	stateStore sm.Store
 
 	blockExec   *sm.BlockExecutor
-	store       *store.BlockStore
+	store       sm.BlockStore
```

Comment:
this piqued my interest, mostly the it becoming not a pointer.

Reply:
Let me add the "story" for this change as a comment here.
When navigating the code back and forth, as I became familiar with the blocksync reactor logic, and the interactions of all reactors with the block store, I realized that all other places were referring to the block store via an interface, rather than using the struct directly, so I introduced this change:

* for consistency with similar places referring to the block store
* to "test" my understanding of the whole thing

Now, AFAICR, this change is not strictly needed and could be reverted if you folks believe it's better to use the struct directly (in that case, I'd like to understand why similar references to the block store are done via interfaces).

Finally, the answer to your "mostly"-prefixed question: the pointer disappears because `sm.BlockStore` is an interface.